src/liballoc/boxed.rs

```diff
@@ -395,11 +395,8 @@ impl<T: Clone> Clone for Box<T> {
 #[stable(feature = "box_slice_clone", since = "1.3.0")]
 impl Clone for Box<str> {
     fn clone(&self) -> Self {
-        let len = self.len();
-        let buf = RawVec::with_capacity(len);
         unsafe {
-            ptr::copy_nonoverlapping(self.as_ptr(), buf.ptr(), len);
-            from_boxed_utf8_unchecked(buf.into_box())
+            from_boxed_utf8_unchecked(self.as_bytes().into())
```

Comment:
But where does this even allocate and copy now...?

Reply:
In the From<[u8]> for Box<[u8]> impl, which is also adjusted in this PR

Reply:
To clarify, the impl I am talking about is the generic one over T

Reply:
Could you call `Box<_>::from(...)` instead of `.into()` to make that clearer? And maybe move that out of the `unsafe` block to make it clear that that is not a weird thing that "steals" `self` or so.

Reply:
good call, fixed